src/Addons/CategoryFetcher.php

```diff
@@ -0,0 +1,240 @@
+                return $category;
+            }
+
+            if (!empty($category['categories'])) {
+                $subCategory = $this->searchCategory($category['categories'], $searchedCategoryId);
+                if (null !== $subCategory) {
+                    return $subCategory;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param array $category
+     *
+     * @return string
+     */
+    private function getDescription(array $category)
+    {
+        $defaultDescription = !empty($this->defaultData['description']) ? $this->defaultData['description'] : '';
+        //Clean link used to fetch description (no need for tracking then)
+        if (empty($category['clean_link'])) {
+            return $defaultDescription;
+        }
+
+        $circuitBreaker = $this->factory->create([
+            'closed' => [self::CLOSED_ALLOWED_FAILURES, self::PLATFORM_TIMEOUT_SECONDS, 0],
+            'client' => [
+                'method' => 'GET',
+            ],
+        ]);
+
+        $categoryResponse = $circuitBreaker->call($category['clean_link'], function () { return false; });
+        if (empty($categoryResponse)) {
+            return $defaultDescription;
+        }
+
+        $cssSelector = new CssSelectorConverter();
+        $document = new DOMDocument();
+        $document->loadHTML($categoryResponse);
+        $xpath = new DOMXPath($document);
+        $descriptionNode = $xpath->query($cssSelector->toXPath('#category_description'))->item(0);
+        $categoryDescription = '';
+        /** @var DOMNode $childNode */
+        foreach ($descriptionNode->childNodes as $childNode) {
+            $categoryDescription .= $childNode->ownerDocument->saveHTML($childNode);
+        }
+
+        return !empty($categoryDescription) ? $categoryDescription : $defaultDescription;
+    }
+
+    /**
+     * Updates link property with a correctly formatted url with tracking parameters
+     *
+     * @param array $category
+     * @param string $isoCode
+     *
+     * @return array
+     */
+    private function addTracking(array $category, $isoCode)
+    {
+        if (empty($category['link'])) {
+            return $category;
+        }
+
+        $parsedUrl = parse_url($category['link']);
+        if (false === $parsedUrl) {
+            return $category;
+        }
+
+        $parameters = [];
+        if (!empty($parsedUrl['query'])) {
+            parse_str($parsedUrl['query'], $parameters);
+        }
+
+        $parameters['utm_source'] = 'back-office';
+        $parameters['utm_medium'] = 'modules';
+        $parameters['utm_campaign'] = 'back-office-' . strtoupper($isoCode);
+
+        //Clean link used to fetch description (no need for tracking then)
+        $category['clean_link'] = self::ADDONS_BASE_URL . $parsedUrl['path'];
+        $category['link'] = self::ADDONS_BASE_URL . $parsedUrl['path'] . '?' . http_build_query($parameters);
```

Comment:
@matks We had issue with the http_build_query and addons api few months ago do you remember? (with modules) :thinking:

Reply:
anyone remembers what was this issue?

Reply:
https://github.com/PrestaShop/PrestaShop/pull/12622#discussion_r258207052